staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -128,6 +147,17 @@ func (c *Config) SkipComplete() completedConfig {
 	return completedConfig{c}
 }
 
+func (r *aggregatorRouting) ResolveEndpoint(namespace, name string) (*url.URL, error) {
+	if r.enabled {
```

Comment:
I would still prefer to see two different implementations rather than one implementation with an if statement.